lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -607,13 +555,25 @@ static void topK(Tensor &outW, Tensor &indW, Tensor &inW, size_t k) {
 void OpenCLFunction::execute(ExecutionContext *context) {
   (void)context;
 
-  {
-    auto ev = context->scopedEvent("alloc");
-    // TODO: deviceBuffer_ should go into DeviceBindings in context.
-    deviceBuffer_ = allocDeviceBuffer(runtimeBundle_.getConstantWeightSize() +
-                                      runtimeBundle_.getMutableWeightSize() +
-                                      runtimeBundle_.getActivationsSize());
-  }
+  auto clBindings = static_cast<runtime::OpenCLDeviceBindings *>(
+      context->getDeviceBindings());
+
+  deviceBuffer_ = clBindings->deviceBuffer;
+  deviceId_ = clBindings->deviceId;
+  commands_ = clBindings->commandQueue;
+  context_ = clBindings->context;
+
+  std::vector<std::string> options;
+  // Configure the kernels by providing the size of size_t on the host size.
+  // This is required to e.g. properly pass struct parameters of types like
+  // ShapeNHWC, ShapeNCHW, etc. The definitions of these types on the host side
+  // use size_t for their members and they should be defined on the OpenCL's
+  // side using integer types of the same width.
+  addIntOption(options, "SIZEOF_HOST_SIZE_T", sizeof(size_t));
+  // Create the program from the source.
+  std::string source(reinterpret_cast<const char *>(kernels_cl_src),
+                     kernels_cl_src_size);
+  createProgram(source, options, commands_);
```

Comment:
This code would "compile" the program using OpenCL compiler on the first call of `execute`. Shouldn't it be done when the network is added to the device manager?